File: api/src/main/java/fr/huiitre/tools/modules/riot/valorant/application/user/command/AddSkinToStoreHistoryCommand.java
package fr.huiitre.tools.modules.riot.valorant.application.user.command;

import java.time.LocalDate;
import java.util.List;

public class AddSkinToStoreHistoryCommand {
    private List<Long> skinIds;
    private LocalDate seenAt;

    public AddSkinToStoreHistoryCommand() {}

    public AddSkinToStoreHistoryCommand(List<Long> skinIds, LocalDate seenAt) {
        this.skinIds = skinIds;
        this.seenAt = seenAt;
    }

    public List<Long> getSkinIds() {
        return skinIds;
    }

    public void setSkinIds(List<Long> skinIds) {
        this.skinIds = skinIds;
    }

    public LocalDate getSeenAt() {
        return seenAt;
    }

    public void setSeenAt(LocalDate seenAt) {
        this.seenAt = seenAt;
    }
}

File: api/Modules/Riot/Valorant/Application/User/Commands/AddSkinToStoreHistoryCommand.cs
namespace Tools.Api.Modules.Riot.Valorant.Application.User.Commands;

// L'ajout est groupé : toute la rotation du jour part en un appel.
//
// Les deux champs sont facultatifs dans le JSON reçu — la liaison ne garantit rien, le use case
// traite l'absence : liste nulle = rien à faire, date nulle = aujourd'hui.
public sealed record AddSkinToStoreHistoryCommand(
    List<long>? SkinIds,
    DateOnly? SeenAt,
    long AccountId
);
